src/Uno.UI/Controls/Window.macOS.cs

```diff
@@ -44,8 +44,210 @@ public Window(CGRect contentRect, NSWindowStyle aStyle, NSBackingStore buffering
+			RegisterForDraggedTypes(new string[]
+				{
+					NSPasteboard.NSPasteboardTypeUrl,
+					NSPasteboard.NSPasteboardTypeTIFF,
+					NSPasteboard.NSPasteboardTypePNG,
+					NSPasteboard.NSPasteboardTypeHTML,
+					NSPasteboard.NSPasteboardTypeRTF,
+					NSPasteboard.NSPasteboardTypeRTFD,
+					NSPasteboard.NSPasteboardTypeFileUrl,
+					NSPasteboard.NSPasteboardTypeString
+
+					/* For future use
+					UTType.URL,
+					UTType.Image,
+					UTType.HTML,
+					UTType.RTF,
+					UTType.FileURL,
+					UTType.PlainText,
+					*/
+				});
+		}
+
+		/// <summary>
+		/// Method invoked when a drag operation enters the <see cref="NSWindow"/>.
+		/// </summary>
+		/// <remarks>
+		/// 
+		/// While it is never documented directly, DraggingEntered can be added to NSWindow just like NSView.
+		/// Key docs telling this story include:
+		/// 
+		///  (1) NSWindow documentation does not include most NSView drag/drop methods
+		///      https://developer.apple.com/documentation/appkit/nswindow
+		///  (2) Since NSWindow documentation doesn't reference them, they also aren't included in Xamarin
+		///      https://docs.microsoft.com/en-us/dotnet/api/appkit.nswindow?view=xamarin-mac-sdk-14
+		///  (3) However, the 'draggingEntered' method docs reference 'Window' at the very end
+		///      "Invoked when the mouse pointer enters the destination’s bounds rectangle (if it is a view object)
+		///       or its frame rectangle (if it is a window object). "
+		///      https://developer.apple.com/documentation/appkit/nsdraggingdestination/1416019-draggingentered
+		///  (4) In legacy macOS docs this is a little more explitly stated:
+		///      https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/DragandDrop/Concepts/dragdestination.html#//apple_ref/doc/uid/20000977-BAJBJFBG
+		///  (5) Other macOS drag/drop articles and blogs refer to dragging/dropping directly from Window - although
+		///      give no examples.
+		/// 
+		/// The Export "method_name" attribute is fundamentally important to make this work, removing it will
+		/// break functionality and the method will never be called by macOS. Again, this seemly is because
+		/// Xamarin.macOS is not aware of it.
+		/// 
+		/// </remarks>
+		/// <param name="info">Information about the dragging session from the sender.</param>
+		/// <returns>The accepted drag operation(s).</returns>
+		[Export("draggingEntered:")] // Do not remove
+		public virtual NSDragOperation DraggingEntered(NSDraggingInfo draggingInfo)
+		{
+			try
+			{
+				return DraggingEnteredAction.Invoke(draggingInfo);
+			}
+			catch
+			{
+				return NSDragOperation.None;
+			}
+		}
+
+		/// <summary>
+		/// Code to execute when the <see cref="DraggingEntered(NSDraggingInfo)"/> method is invoked.
+		/// This can be used in a similar way to <see cref="UIElement.DragEnter"/>.
+		/// </summary>
+		public Func<NSDraggingInfo, NSDragOperation> DraggingEnteredAction { get; set; } =
```

Comment:
I actually decided against using events because:
 - There is nothing stopping more than one object from subscribing to an event, that wouldn't make sense in this case.
 - I need returned information -- also very strange with an event although it could be done with a property in a new eventArgs class. 
 - This design more closely follows how macOS actually works. It's invoking a method to be handled internally -- not notifying any observers to take appropriate action. Fundamentally, it is a different concept from what we would normally see in c#/UWP so I implemented as such.

Reply:
I can switch to using a good old-fashioned delegate if it's needed :) I wasn't aware of any performance concerns with Func or Action usage though. In fact, these are widely used now days and recommended over the original delegate in cases like this, easier to read and what not. Is there any information you can share about those performance concerns? Is that something that only is a concern in an AOT scenario? I also know Uno itself uses these all over the place so wonder what the criteria is for decided where to use them?

Reply:
Yeah I would agree that writing our own delegate is no longer the norm. It's something that we discovered while doing some performance passes and which is specific to Xamarin / mono in AOT mode. For more details we gonna have to ask to the master of the subject: @jeromelaban 🤓. But for now as the rule is to use "old-fashioned delegate", could you please change them?

I've no real concern of not using an event as they are indeed not intended to be listen by application, but in that case I would make those properties `internal`. But even then, as it's in the `Uno` namespace, it's just a suggestion!

Reply:
Yea, even the Microsoft docs say not to create typed delegates for every signature (code bloat) and instead use Func and Action these days. If this is some AOT issue with generics that does make sense to me. Seems like a Xamarin issue should be filed as well though.

I'll change the code hopefully by end of today.